### ISSUE TITLE
Controller doesn't start with ipv6 disabled

### ISSUE THREAD

**Issue by kjchmielewski** [haproxytech/kubernetes-ingress#128]
Hi 
I'm trying to deploy helm chart as follows:

``$ helm install haproxy-ingress haproxytech/kubernetes-ingress --set controller.kind=DaemonSet``

What happens is that controller doesn't start, here are the logs:

```2019/12/20 07:36:11 HAProxy Ingress Controller v1.2.5 828787a

2019/12/20 07:36:11 Build from: git@github.com:haproxytech/kubernetes-ingress.git
2019/12/20 07:36:11 Build date: 2019-11-18T09:29:59

2019/12/20 07:36:11 ConfigMap: ingress-nginx/haproxy-ingress-kubernetes-ingress
2019/12/20 07:36:11 Ingress class: haproxy
2019/12/20 07:36:11 main.go:82: Default backend service: ingress-nginx/haproxy-ingress-kubernetes-ingress-default-backend
2019/12/20 07:36:11 main.go:83: Default ssl certificate: ingress-nginx/haproxy-ingress-kubernetes-ingress-default-cert
2019/12/20 07:36:11 controller.go:92: Running with  HA-Proxy version 2.0.8 2019/10/23 - https://haproxy.org/
2019/12/20 07:36:11 controller.go:97: Starting HAProxy with /etc/haproxy/haproxy.cfg
2019/12/20 07:36:11 controller.go:110: Running on k8s-node02
[WARNING] 353/073611 (20) : Can't open server state file '/var/state/haproxy/global': No such file or directory
2019/12/20 07:36:11 controller.go:69: Running on Kubernetes version: v1.16.2 linux/amd64
[ALERT] 353/073611 (20) : Starting frontend https: cannot create listening socket [:::443]
[ALERT] 353/073611 (20) : Starting frontend http: cannot create listening socket [:::80]
```
I cannot find an option for disabling ipv6 binding.
I have ipv6 disabled in kernel with option ipv6.disable=1

**Comment by bedis** [comment]
hi, this looks like ports 80 and 443 are already bound by somebody else.
`DaemonSet` automatically enables `hostNetwork`.
Can you go on the host and run `ss -lpten | fgrep 80` (and same for 443). If `ss`is not available, use this old good `netstat`.
you should see that those ports are already bound by somebody else.

**Comment by kjchmielewski** [comment]
Thanks for Your advice,
I run commands as per instructions, here's what I get:

> [root@k8s-node01 ~]# ss -lpten | fgrep 80
LISTEN     0      128    10.59.58.54:10050                    *:*                   users:(("zabbix_agentd",pid=2114,fd=4),("zabbix_agentd",pid=2113,fd=4),("zabbix_agentd",pid=2112,fd=4),("zabbix_agentd",pid=2111,fd=4),("zabbix_agentd",pid=2110,fd=4),("zabbix_agentd",pid=2109,fd=4)) uid:997 ino:19974 sk:ffff92af5ea68f80 <->
LISTEN     0      128          *:9100                     *:*                   users:(("node_exporter",pid=28466,fd=3)) uid:65534 ino:41450424 sk:ffff92af61bbcd80 <->
LISTEN     0      128          *:10256                    *:*                   users:(("kube-proxy",pid=28601,fd=11)) ino:792225 sk:ffff92af61bbae80 <->
LISTEN     0      128          *:22                       *:*                   users:(("sshd",pid=1668,fd=3)) ino:23646 sk:ffff92af61bb8000 <->
LISTEN     0      128    127.0.0.1:43383                    *:*                   users:(("kubelet",pid=27899,fd=16)) ino:782332 sk:ffff92af603a2e80 <->
[root@k8s-node01 ~]# ss -lpten | fgrep 443

it looks like there is no other process bound to ports 80 and 443.
So I deleteted deployment

``$helm delete haproxy-ingress --purge``

and installed without _--set controller.kind=DaemonSet_ option.
Effect seems to be the same. I provide full log (previous was truncated)
[kubectl_logs_haproxy-ingress-kubernetes-ingress-8cd69dd-rbhf7.txt](https://github.com/haproxytech/kubernetes-ingress/files/3987548/kubectl_logs_haproxy-ingress-kubernetes-ingress-8cd69dd-rbhf7.txt)

**Comment by Mo3m3n** [comment]
Hi
> I cannot find an option for disabling ipv6 binding.
> I have ipv6 disabled in kernel with option ipv6.disable=1
I am suspecting that you're having these errors because HAProxy is trying to bind to ipv6 and you have disabled ipv6 in the kernel.
Can you enable ipv6 in kernel and try again ?
Currently there is no option to disable ipv6, we can add this option in few days.

**Comment by kjchmielewski** [comment]
Hi

Sorry for an delay but I have to wait until I could reconfigure kernel parameters on vm running as k8s node.
I can confim that with ipv6 enabled in kernel HAProxy starts and runs properly.

```
$ kubectl logs haproxy-ingress-kubernetes-ingress-57c67c7f78-5xs72 
2020/01/01 17:43:13 
 _   _    _    ____                                            
| | | |  / \  |  _ \ _ __ _____  ___   _                       
| |_| | / _ \ | |_) | '__/ _ \ \/ / | | |                      
|  _  |/ ___ \|  __/| | | (_) >  <| |_| |                      
|_| |_/_/   \_\_|   |_|  \___/_/\_\\__, |                      
 _  __     _                       |___/             ___ ____  
| |/ /   _| |__   ___ _ __ _ __   ___| |_ ___  ___  |_ _/ ___| 
| ' / | | | '_ \ / _ \ '__| '_ \ / _ \ __/ _ \/ __|  | | |      
| . \ |_| | |_) |  __/ |  | | | |  __/ ||  __/\__ \  | | |___ 
|_|\_\__,_|_.__/ \___|_|  |_| |_|\___|\__\___||___/ |___\____| 
                                                                                                                         

2020/01/01 17:43:13 HAProxy Ingress Controller v1.2.5 828787a

2020/01/01 17:43:13 Build from: git@github.com:haproxytech/kubernetes-ingress.git
2020/01/01 17:43:13 Build date: 2019-11-18T09:29:59

2020/01/01 17:43:13 ConfigMap: haproxytech/haproxy-ingress-kubernetes-ingress
2020/01/01 17:43:13 Ingress class: haproxy
2020/01/01 17:43:13 main.go:82: Default backend service: haproxytech/haproxy-ingress-kubernetes-ingress-default-backend
2020/01/01 17:43:13 main.go:83: Default ssl certificate: haproxytech/haproxy-ingress-kubernetes-ingress-default-cert
2020/01/01 17:43:13 controller.go:92: Running with  HA-Proxy version 2.0.8 2019/10/23 - https://haproxy.org/
2020/01/01 17:43:13 controller.go:97: Starting HAProxy with /etc/haproxy/haproxy.cfg
2020/01/01 17:43:13 controller.go:110: Running on haproxy-ingress-kubernetes-ingress-57c67c7f78-5xs72
2020/01/01 17:43:13 controller.go:69: Running on Kubernetes version: v1.16.2 linux/amd64
[WARNING] 000/174313 (25) : Can't open server state file '/var/state/haproxy/global': No such file or directory
[NOTICE] 000/174313 (26) : New worker #1 (27) forked
2020/01/01 17:43:20 controller-haproxy.go:144: HAProxy reloaded
[NOTICE] 000/174320 (40) : New worker #1 (41) forked
```

**Comment by kungfu71186** [comment]
Have the same exact issue. When ipv6 is disabled, haproxy won't start.


```
kubectl get pods

NAME                                                              READY   STATUS    RESTARTS   AGE
haproxy-ingress-kubernetes-ingress-default-backend-d859d992hfbn   1/1     Running   0          83s
haproxy-ingress-kubernetes-ingress-default-backend-d859d99gzr62   1/1     Running   0          83s
haproxy-ingress-kubernetes-ingress-r7h9k                          1/1     Running   0          83s
haproxy-ingress-kubernetes-ingress-sdb79                          0/1     Running   2          83s
haproxy-ingress-kubernetes-ingress-v79wq                          0/1     Running   2          83s

```

The pod that is running is the node that has ipv6 enabled. The other two do not have ipv6 enabled.

**Comment by oktalz** [comment]
we plan to add option to remove ipv6 (or ipv4) binds
i expect this feature to land in master probably next week

**Comment by kungfu71186** [comment]
@oktalz oh sweet. Yeah i just found why its happening, saw the config.

**Comment by rmohammed-xtime** [comment]
@oktalz Was this added? I don't see support for disabling ipv6 in 1.4.5

**Comment by matclayton** [comment]
Any update on this ticket, we're also seeing the same issue.

**Comment by prometherion** [comment]
The feature has been implemented with db51a5dbb6a66ee4959730a96187b3fb141b5906 and planned for the next upcoming release (`1.5.0`)

**Comment by metsuke0** [comment]
Is there confirmation that this issue will be fixed by 1.5.0? Is there a flag that needs to be passed in order for it to work? I tried 1.5.0-rc1 with no modifications and it is still trying (and failing) to bind to ipv6 ports.

**Comment by prometherion** [comment]
> Is there a flag that needs to be passed in order for it to work?

Just need to add the flag `--disable-ipv6` in the Pod args: we're not disabling it by default to prevent breaking changes.
